functorch/experimental/_cond.py

```diff
@@ -206,14 +208,17 @@ def _has_potential_branch_input_alias(branch, inputs):
 
     input_storages = set()
     for node in gm.graph.nodes:
-        if node.op == "placeholder":
+        # Scalars don't have node.meta
+        if node.op == "placeholder" and "val" in node.meta:
             input_storages.add(StorageWeakRef(node.meta['val']._typed_storage()))
         if node.op == "output":
-            for out in node.args:
-                out_storage = StorageWeakRef(out.meta["val"]._typed_storage())
-                if out_storage in input_storages:
-                    return True
-
+            def check_alias(out):
+                if "val" in out.meta:
+                    out_storage = StorageWeakRef(out.meta['val']._typed_storage())
```

Comment:
This is a preexisting condition, but seeing this was quite shocking to me. What are you testing typed storage on the fake tensor vals for?